editor/app/components/slider.jsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import "./slider.css";
+
+/**
+ *
+ * @param {number} start - The start value of the range.
+ * @param {number} end - The end value of the range.
+ * @param {number} step - The increment given to each step.
+ * @returns An array of numbers from start to end with increments based on step
+ */
+const range = (start, end, step) =>
+  Array.from({length: (end - start) / step + 1}, (_, i) => start + i * step);
+
+/**
+ * A custom slider component that extends the standard input range control.
+ */
+export default class Slider extends React.Component {
+  labels = [];
+
+  constructor(props) {
+    super(props);
+    const {min, max, step} = this.props;
+
+    // Create an array of labels for the slider here so these don't have to be
+    // recalculated on every render.
+    this.labels = range(min, max, step);
```

Comment:
oh I was thinking that we can convert this component to be a render function component instead of a class, and use the memo hook to calculate this labels values only once.  let me know what you think.

Reply:
ha! good call... I was just mentioning this in a previous comment, will do that in a moment.